daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/ChoiceCombinator.scala

```diff
@@ -257,9 +267,8 @@ case class ChoiceCombinator(ch: ChoiceTermBase, alternatives: Seq[Gram])
     if (mapValues.isEmpty)
       new NadaUnparser(null)
     else {
-      val firstBranchUnparser = alternatives(0).unparser
       new ChoiceCombinatorUnparser(ch.modelGroupRuntimeData, eventUnparserMap,
-        choiceLengthInBits, optDefaultableBranchUnparser, firstBranchUnparser)
+        choiceLengthInBits, optDefaultableBranchUnparser)
```

Comment:
Before committing this, we really must test mil-std-2045 and vmf. These have been turning up many problems related to choices for me.

Reply:
Both have been tested and are passing

Reply:
+1 then.